src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts

```diff
@@ -0,0 +1,166 @@
+import { Component, Input } from "@angular/core";
+import { GlobalVarsService } from "../../global-vars.service";
+import { ActivatedRoute, Router } from "@angular/router";
+import { Datasource, IAdapter, IDatasource } from "ngx-ui-scroll";
+import { BackendApiService, PostEntryResponse, ProfileEntryResponse } from "../../backend-api.service";
+import * as _ from "lodash";
+
+@Component({
+  selector: "diamond-posts",
+  templateUrl: "./diamond-posts.component.html",
+  styleUrls: ["./diamond-posts.component.sass"],
+})
+export class DiamondPostsComponent {
+  constructor(
+    public globalVars: GlobalVarsService,
+    private router: Router,
+    private backendApi: BackendApiService,
+    private route: ActivatedRoute
+  ) {
+    this.route.params.subscribe((params) => {
+      this.receiverUsername = params.receiver;
+      this.senderUsername = params.sender;
+    });
+  }
+
+  receiverUsername: string;
+  senderUsername: string;
+
+  receiverProfileEntryResponse: ProfileEntryResponse;
+  senderProfileEntryResponse: ProfileEntryResponse;
+
+  datasource: IDatasource<IAdapter<any>> = this.getDatasource();
+  loadingFirstPage = true;
+  loadingNextPage = false;
+  pagedKeys = {
+    0: "",
+  };
+
+  lastDiamondLevelOnPage = {
+    0: 0,
+  };
+
+  pagedRequests = {
+    "-1": new Promise((resolve) => {
+      resolve([]);
+    }),
+  };
+
+  lastPage = null;
+
+  static PAGE_SIZE = 10;
+
+  getDatasource() {
```

Comment:
we should really make this infinite scroller component reusable. don't let this block merge though

Reply:
agreed it's on my list.